.github/workflows/ci.yml

```diff
@@ -0,0 +1,97 @@
+name: CI
+
+on:
+  pull_request:
+    types: [opened, synchronize]
```

Comment:
The one thing I was least sure about is when we want this to be run. What I currently have it set to is for this workflow to be triggered whenever a pull request is created (hence the the `opened`) and whenever the PR branch changes relative to the base branch (i.e., if something is merged to the base branch, or this branch is updated - hence the `synchronize`).

I have it set this way because it seemed the best way to ensure that the branch can be merged into main without breaking anything (for example, if we only re-ran the test on change to the PR branch itself and not also the base branch, then it could be that another change is merged into the base branch which is incompatible with the changes in the PR branch).

Though, I'm not totally clear on the protocol we want to follow for testing and merging, so I'd appreciate input.

Reply:
Discussed in person - this is likely what we want

1) We aren't paying for compute here, and frequency of updates to main or PR branches isn't super high, so not too worried about this triggering too many jobs or being too noisy
2) We aren't trying to guarantee master stays green with this (because of the copybara merge cycle).  However, I think the combination of triggers here along with testing on the merge commit would guarantee that if we merged straight from PRs here into symforce main, which is nice (assuming `synchronize` does what we think and retriggers any time the PR HEAD or base branches move or change)

Noting we also talked about adding a workflow_dispatch trigger here so that tests can be retriggered with a button.  Decided that'd be nice-to-have but not required - also noting that we might need some work to get it to run on the merge commit for a branch as opposed to the HEAD, since `actions/checkout@v3` as configured is checking out the [event ref](https://github.com/actions/checkout#checkout-pull-request-head-commit-instead-of-merge-commit), which is the merge commit for a [pull_request event](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#pull_request), but won't be for a workflow_dispatch event